lib/portlayer/portlayer.go

```diff
@@ -83,5 +84,46 @@ func Init(ctx context.Context, sess *session.Session) error {
 		return err
 	}
 
+	unbindSerialPorts(sess)
+
 	return nil
 }
+
+// unbindSerialPorts disconnects serial ports backed by network on the VCH's old IP
+// for running containers. This is useful when the appliance or the portlayer restarts
+// and the VCH has a new IP. Any errors are logged and portlayer init proceeds as usual.
+func unbindSerialPorts(sess *session.Session) {
+	// Get all running containers from the cache
+	runningState := new(exec.State)
+	*runningState = exec.StateRunning
+	containers := exec.Containers.Containers(runningState)
+
+	for i := range containers {
+		if containers[i].ExecConfig != nil {
+			log.Infof("unbinding serial port for running container: %s", containers[i].ExecConfig.ID)
+		}
+
+		// Obtain a container handle
+		handle := containers[i].NewHandle(context.Background())
+		if handle == nil {
+			log.Error("unable to obtain a handle for container")
```

Comment:
I currently have

```
log.Infof("unbinding serial port for running container: %s", containers[i].ExecConfig.ID)
```

at the start of the every loop iteration. This prints the container ID for every container before operating on it, while avoiding a nil check for `container.ExecConfig` every time we want to log a message. `container.ExecConfig` will be non-nil in most circumstances, but I'm not sure if we can always guarantee that - specially since this code kicks in when the appliance or PL crashes. 

If we're sure that `container.ExecConfig` will always be non-nil, we can simply log the ID every time. Otherwise, I can add a nil-check before every log or do something like:
```
var containerID string
if container.ExecConfig != nil {
  containerID = container.ExecConfig.ID
}
...
log.Errorf("... %s", containerID)
```
What do you think?